### PR TITLE
move main config api to openshift/api

### PR DESCRIPTION
**Pull request by deads2k**
This reacts to the changes in https://github.com/openshift/api/pull/95

/hold

hold until 3.11 stops tracking master

/assign @mfojtik @sttts

### CODE REVIEW

File: vendor/github.com/openshift/api/legacyconfig/v1/serialization.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package v1
```
Comment (sttts) [inline vendor/github.com/openshift/api/legacyconfig/v1/serialization.go]: is the upstream PR open?

File: pkg/cmd/server/apis/config/v1/defaults.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
+package v1
```
Comment (sttts) [inline pkg/cmd/server/apis/config/v1/defaults.go]: this was just split out from conversions.go? Any change?

File: pkg/cmd/server/apis/config/v1/conversions.go [hunk 3]
```diff
@@ -1,409 +1,262 @@
...
-			if out.DisabledAPIGroupVersions == nil {
-				out.DisabledAPIGroupVersions = map[string][]string{}
-			}
+	return nil
+}
 
-			// the APILevels (whitelist) needs to be converted into an internal blacklist
-			if len(in.APILevels) == 0 {
-				out.DisabledAPIGroupVersions[internal.APIGroupKube] = []string{"*"}
+func Convert_config_AdmissionPluginConfig_To_v1_AdmissionPluginConfig(in *internal.AdmissionPluginConfig, out *legacyconfigv1.AdmissionPluginConfig, s conversion.Scope) error {
```
Comment (sttts) [inline pkg/cmd/server/apis/config/v1/conversions.go]: thank you github for making this impossible to review.

Comment (sttts) [reply pkg/cmd/server/apis/config/v1/conversions.go]: Are there any changes in the conversions?

File: vendor/github.com/openshift/api/legacyconfig/v1/serialization.go (hunk 1 above)
Comment (deads2k) [reply vendor/github.com/openshift/api/legacyconfig/v1/serialization.go]: > is the upstream PR open?

Yeah.

File: pkg/cmd/server/apis/config/v1/defaults.go (hunk 2 above)
Comment (deads2k) [reply pkg/cmd/server/apis/config/v1/defaults.go]: > this was just split out from conversions.go? Any change?

no change

File: pkg/cmd/server/apis/config/v1/conversions.go (hunk 3 above)
Comment (deads2k) [reply pkg/cmd/server/apis/config/v1/conversions.go]: > Are there any changes in the conversions?

Yeah. The last four are new. They are used to call a conversion from internal to external on contained runtime.Objects.

**Comment by mfojtik** [comment]
lgtm

**Comment by deads2k** [comment]
updated for real bump,  reapplying label.

Leaving the hold.  when we get word, we'll remove it.

**Comment by deads2k** [comment]
word from derek is that master is open for 4.0.  unholding.

**Pull request merged**